### PR TITLE
Remove autoconf build system

### PR DESCRIPTION
**Pull request by TheMarex**
# Issue

This implements https://github.com/valhalla/valhalla/issues/1299 which might be blocked by https://github.com/valhalla/packaging/issues/17

## Tasklist

 - [x] review - you must request approval to merge any PR to master

### CODE REVIEW

**Comment by TheMarex** [comment]
I'm not sure why it still tries to run the autoconf build job. Looks like the lint job also needs to be ported to cmake.

**Comment by kevinkreiser** [comment]
before merging this can we make sure that we indeed have a tag of valhalla with both the cmake and autotools builds systems? it will make testing the ppa generation stuff easier when i get time to do that on my own.

**Comment by TheMarex** [comment]
Unblocked this by tagging https://github.com/valhalla/valhalla/releases/tag/2.6.0

File: README.md [hunk 1]
```diff
@@ -103,25 +103,20 @@ After getting the dependencies install it with:
...
+cmake .. -DCMAKE_BUILD_TYPE=Release
+make -j$(nproc)
 sudo make install
-# Note: on macOS, a few tests involving time formatting will fail: narrativebuilder, util_odin, mapmatch
 ```
 
-Please see `./configure --help` for more options on how to control the build process. There are a few notable options that you might want to try out:
+For more build options run the interactive GUI:
 
-* `--enable-data-tools=no` will disable building any of the components (library bits, executables and tests) which can be used to create the data that the services run on. This can be useful in embedded situations where you really don't need some of the dependencies above.
```
Comment (oxidase) [inline README.md]: I think it is better to keep the section and only change configure arguments to:
-DENABLE_DATA_TOOLS=OFF
 -DENABLE_PYTHON_BINDINGS=OFF
 -DENABLE_SERVICES=OFF
-DBUILD_SHARED_LIBS=Off

File: README.md [hunk 2]
```diff
@@ -174,19 +169,7 @@ We highly encourage running and updating the tests to make sure no regressions h
 
     make check
 
-You can also build a test coverage report. This requires that the packages `lcov`, `gcov` and `genhtml` be installed. On Ubuntu you can get these with:
-
-    sudo apt-get install lcov
-
-To make the coverage report, configure the build for it:
-
-    ./configure --enable-coverage
```
Comment (oxidase) [inline README.md]: It is also possible to make local code coverage reports
`cmake .. -DENABLE_COVERAGE=On`

and `make coverage` below

**Comment by TheMarex** [comment]
@oxidase Thanks for the review! I adjusted the readme for your recommendations, please take a look again.

**Pull request merged**